kolibri/plugins/learn/assets/src/actions.js

```
@@ -85,30 +102,62 @@ function _getCurrentChannel() {
           }
         }
         resolve(currentChannelId);
+      })
+      .catch((error) => {
+
       });
   });
 }
 
 
+/*
+ * Returns a promise that fetches the the root topic id of the current channel.
+ */
+function _getCurrentChannelRootTopicId() {
+  let currentChannelRootTopicId = null;
+  return new Promise((resolve, reject) => {
+    const currentChannelIdPromise = _getCurrentChannelId();
+    const channelListPromise = _getChannelList();
+    Promise.all([currentChannelIdPromise, channelListPromise])
+    .then(([currentChannelId, channelList]) => {
+      if (currentChannelId && channelList) {
+        for (const channel in channelList) {
+          if (channelList[channel].id === currentChannelId) {
+            currentChannelRootTopicId = channelList[channel].root_pk;
+            break;
+          }
+        }
+      }
+      resolve(currentChannelRootTopicId);
+    })
+    .catch((error) => {
+
+    });
```

empty catch blocks are risky because they silently swallow errors